dask_jobqueue/core.py

```diff
@@ -406,6 +407,6 @@ def _del_pending_jobs(self):
             del self.pending_jobs[job_id]
         return jobs
 
-    def _job_id_from_submit_output(self, out):
-        raise NotImplementedError('_job_id_from_submit_output must be implemented when JobQueueCluster is '
-                                  'inherited. It should convert the stdout from submit_command to the job id')
+    @staticmethod
+    def _job_id_from_submit_output(out):
+        return re.findall(r'\d+', out)[0]
```

Comment:
A few comments about this:
* I am in favour of using a class variable for the regexp pattern. This way the regexp pattern can be tweaked more easily in derived classes. It can also be tweaked in the case where the cluster config is a bit special if needed:
```py
cluster = MyCluster(...)
cluster.job_id_regexp = my_special_regexp
```
* I think the best way is to have a regexp with a named match: r`(?<job_id>:\d+)` by default. This way you can use the name of the match which is more robust:
```py
match = re.search(self.job_id_regexp, out)
job_id = match.groupdict.get('job_id')
```
* There should be better error treatment
```py
if job_id is None:
    raise ValueError('Could not parse job id from submission command output. Job id regexp is {}, submission command output is: {}'.format(self.job_id_regexp, out)
```